java/core/src/main/java/com/whylogs/core/metrics/BaseMetric.java

```diff
@@ -0,0 +1,13 @@
+package com.whylogs.core.metrics;
+
+import lombok.NonNull;
+
+public abstract class BaseMetric<TSubclass extends BaseMetric> extends Metric {
+
+    public BaseMetric(@NonNull String namespace) {
+        super(namespace);
+    }
+
+    public abstract TSubclass merge(TSubclass other);
+    public abstract TSubclass zero(MetricConfig config);
```

Comment:
I think `zero` needs to be static because it's used like a constructor/factory.

Reply:
Moved to static, but I may look at this more with the factory in the next iteration.